### ISSUE TITLE
Declarative Configuration - Ability to control which contact is associated with emitted task

### ISSUE THREAD

**Issue by kennsippell** [medic/medic-conf#205]
Tasks which are emitted by app_settings configuration contain a contact attribute which controls which contact is associated with the task. In the declarative configuration system, the contact associated with the task is the owner of the report that generated the task. [Here](https://github.com/medic/medic-conf/blob/master/src/nools/lib.js#L142) 

For Muso's mRDT supervisor workflows, we need to generate tasks for supervisors which are based on the content of reports owned by patients. 

More details contained in: https://github.com/medic/medic-projects/issues/6519

**Comment by kennsippell** [comment]
@garethbowen @abbyad @MaxDiz This is impacting the ongoing mRDT trials and should be considered for inclusion in 3.6 or 3.7.

@senseibara @Fatoufall

**Comment by garethbowen** [comment]
I'll schedule this for 3.7.0. medic-conf is on a separate release schedule so we can release earlier, but I don't want this to block 3.6.0. Make sense?

**Comment by Fatoufall** [comment]
@garethbowen from our perspective 3.6 should allow us to scale up mRDT and this issue has an impact on this scale up. Is there any way to consider this in 3.6?

**Comment by kennsippell** [comment]
@Fatoufall, even in 3.7 I can prioritise this to be done soon.  With medic-conf issues, we can do a release whenever the work is ready.

**Comment by Fatoufall** [comment]
great and thanks!

**Comment by dianabarsan** [comment]
Ready for AT on `205-dec-contact`.

**Comment by kennsippell** [comment]
Please don't merge this quite yet, due to some feedback recieved when reviewing the PR in medic-docs. https://github.com/medic/medic-docs/pull/122

**Comment by kennsippell** [comment]
Ready for AT

**Comment by newtewt** [comment]
@kennsippell  I'm noticing an issue with this change. I see 1 task created for the contact based on the task config below.  When I look at the clinic that was created today there is no task displayed in the tasks list. When I look at the other clinics they show the task for this clinic. Using  medic-conf version 2.0.4 the opposite happens. The task displays on the clinic that was created and the other clinics don't see that task. 
```

{
    name: 'task',
    title: 'foo',
    actions: [{ form: 'example' }],
    contactLabel: function(c){
      return c.contact._id;
    },
    appliesToType: 'clinic',
    events: [{ days: 10, start: 100, end: 1 }],
    appliesTo: 'contacts',
    appliesIf: function() {
      return true;
    },
    resolvedIf: function() {
      return false;
    }
  }

```

**Comment by kennsippell** [comment]
@newtewt I'm having trouble reproducing this difference. Would you be explicit with the clinics you have created, their reported_dates, and the tasks you're seeing?

**Comment by newtewt** [comment]
I can actually see the issue when just using `contactLabel` and not using it.  When I use the same task as shown above and remove `contactLabel`. I see no tasks on clinic_1.

**Comment by kennsippell** [comment]
Thanks for clarifying! Great bug.

**Comment by kennsippell** [comment]
@newtewt I tested this on 3.6.1 and I think things are working as expected (when we set a custom contactLabel, it doesn't appear in any of the contact pages). I've opened https://github.com/medic/medic#5899 to fix this in 3.7.

**Comment by newtewt** [comment]
@kennsippell  cool, I'll close this as that was the only outstanding issue.
